### DISCUSSION TITLE
chatGPTを用いたPHPへの翻訳

### DISCUSSION THREAD

**Discussion by PrefKarafuto** [PrefKarafuto/ex0ch#42] [General]
PHPへの書き換えに関する事項はここで扱います

**Comment by PrefKarafuto** [comment]
どういう感じでchatGPTに投げて翻訳していくのがいいんでしょうかね

**Reply by Ematojp** [reply]
PerlからPHPはまだ本格的にやっていないですけどPHPのだけなら”脆弱性とコーディングミスを発見して修正してください。また原型を可能な限り保ちながらPHP8.2で非推奨とされている関数があれば別の安全なものに置き換えてください。”って感じで命令してます
多分ちゃんとしたチェックはできてませんが初歩的なXSSやインジェクション攻撃対策になるくらいには修正してくれます

**Reply by PrefKarafuto** [reply]
ソースファイル毎に翻訳してます？それとも関数で分割したりしますか？
chatGPTのプランにもよるとは思いますが、自分の環境だとあまりに長いものは受け付けてくれませんので

**Reply by Ematojp** [reply]
コメントと関数でざっくりと分割してます
ものによって品質の差がありますが中途半端なものを入力してもちゃんと出力することが多いです
関数でも長すぎたりHTMLとPHPが混在しているものは低品質な出力が多いです
単体の言語であれば150行以下でしたら正しく解釈して出力することが多い印象です
もちろんプログラミング知識ある人が指示したらより長く複雑なものでもより高品質な出力が得られると思います

**Reply by PrefKarafuto** [reply]
3.5と4どっち使ってますか？

**Reply by Ematojp** [reply]
基本4ですね

**Reply by PrefKarafuto** [reply]
となると、chatGPT関係はお任せすることになりそです

**Reply by PrefKarafuto** [reply]
あとコラボレーターに招待いたしました

**Reply by PrefKarafuto** [reply]
GPTで翻訳する際に一緒に注釈もソース内にコメントの形で出力することは可能でしょうか

**Reply by Ematojp** [reply]
一旦コメント追加しました
一般的なコメントから初学者向け等アレンジできますがどの程度のコメントが必要になりますか？

**Reply by PrefKarafuto** [reply]
とりあえずは一般的なコメントと、あとモジュールのPerlからPHPへの翻訳にあたっての留意事項などがあれば付加していただきたいです。

**Reply by PrefKarafuto** [reply]
翻訳が終わったファイルは順次UPしていただいて構いません

**Comment by PrefKarafuto** [comment]
そういえば以前協力してくださった草井さんって連絡とれますかね
ブログに連絡先が見当たらなかったので

**Reply by Kusaimara** [reply]
お久しぶりです
メール返信しました

**Reply by PrefKarafuto** [reply]
お久しぶりです
ご多忙のなか恐縮ですが、もしお時間が空いたなら是非ご助力の程、ご一考ください

**Reply by Kusaimara** [reply]
分かりました
今は忙しいですが時間を見つけてお手伝いします

**Reply by Kusaimara** [reply]
すみません、メールを送信しました
ご確認いただければ幸いです

**Comment by PrefKarafuto** [comment]
翻訳したソースファイルを置く場所を作成しました。
https://github.com/PrefKarafuto/New_0ch_Plus/tree/main/PHP

**Reply by PrefKarafuto** [reply]
確認のため試しに何かUPしてみてください

**Reply by Ematojp** [reply]
GPT-4でbbsmenu.cgiを翻訳してアップロードしました

**Reply by PrefKarafuto** [reply]
確認しました
きちんと見えています

**Comment by PrefKarafuto** [comment]
いつから翻訳を開始しましょうか
現時点でスレ主用コマンドの実装が途中までになっているので、PHPで新しく効率的に書き直したほうがいいですかね

**Reply by PrefKarafuto** [reply]
試しにbbsmenu.cgiを動かしてみて、これは速く翻訳に取り掛かったほうがいいかと思います。
他のclassが翻訳されていないので動くか確認しようがないですからね
